src/dotnet-install.sh

```diff
@@ -1001,18 +1001,34 @@ downloadcurl() {
     local remote_path_with_credential="${remote_path}${feed_credential}"
     local curl_options="--retry 20 --retry-delay 2 --connect-timeout 15 -sSL -f --create-dirs "
     local failed=false
+    local curl_exit_code=0;
     if [ -z "$out_path" ]; then
-        curl $curl_options "$remote_path_with_credential" 2>&1 || failed=true
+        curl $curl_options "$remote_path_with_credential" 2>&1
+        curl_exit_code=$?
+        if [ $curl_exit_code -gt 0 ]; then
+            failed=true
+        fi
     else
-        curl $curl_options -o "$out_path" "$remote_path_with_credential" 2>&1 || failed=true
+        curl $curl_options -o "$out_path" "$remote_path_with_credential" 2>&1
+        curl_exit_code=$?
+        if [ $curl_exit_code -gt 0 ]; then
+            failed=true
+        fi
     fi
-    if [ "$failed" = true ]; then
-        local disable_feed_credential=false
-        local response=$(get_http_header_curl $remote_path $disable_feed_credential)
-        http_code=$( echo "$response" | awk '/^HTTP/{print $2}' | tail -1 )
+    
+    if [[ "$failed" = true ]]; then
         download_error_msg="Unable to download $remote_path."
-        if  [[ $http_code != 2* ]]; then
+        # Check for curl timeout codes
+        if [[ $curl_exit_code == 7 || $curl_exit_code == 28 ]]; then
+            http_code="408"
             download_error_msg+=" Returned HTTP status code: $http_code."
```

Comment:
I wonder if this will cause any confusion when diagnosing issues. After all, `curl(7)` and `curl(28)` are not the same as `http(408)`.

It is my understanding that timeout issues from curl mean that it has failed to reach the server (which points to a network issue) while http 408 means that the server has knowingly terminated the connection (which means network is fine).

Therefore, I suggest we leave the `http_code` untouched while updating the error msg
```suggestion
            download_error_msg+=" Failed to reach the server: connection timeout."
```

Reply:
Same concerns for `wget` error code `4`.

Reply:
Agree!